### PR TITLE
RHMAP-15494 - update template as part of node 6 update

### PR DESCRIPTION
**Pull request by rachael-oregan**
## Motivation
Verify that the Hello World Cloud Cluster App could be built with node 6, that the tests passed and that the readme was up to date.

## Result

- Included Corinne's changes to swap out turbo for mocha test runner => [remove turbo](https://github.com/feedhenry-templates/helloworld-cloud-cluster/pull/58)

- Updated the gruntfile.

- Updated the npm scripts in `package.json` and the `devDependencies` and `dependencies`

- Removed node 0.10 and added node 6 to `travis.yml`

- Updated `application.js` to check what is tying to start the server - if it's mocha then skip cluster code and run `startWorker()`

- Updated the readme to reflect the changes.

## Jira
https://issues.jboss.org/browse/RHMAP-15494

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ module.exports = function(grunt) {
       }
     },
     concurrent: {
-      serve: ['nodemon', 'watch'],
+      serve: ['nodemon'],
```
Comment (jcesarmobile) [inline Gruntfile.js]: I don't think we should remove the watch, other templates have it and can be useful.

File: lib/hello.js [hunk 2]
```diff
@@ -5,7 +5,11 @@ var cors = require('cors');
 function helloRoute() {
   var hello = new express.Router();
   hello.use(cors());
-  hello.use(bodyParser());
+  //hello.use(bodyParser());
```
Comment (jcesarmobile) [inline lib/hello.js]: remove the comment.
This change is for removing the deprecate warnings? In other templates we didn't update body-parser so we didn't encounter the deprecate warnings and I was going to create a new issue about updating templates for using latest dependencies

File: application.js [hunk 3]
```diff
@@ -108,7 +109,8 @@ function workerExitHandler(worker, code, signal) {
 // Start function
 // The number of workers to start can be specified with the FH_NUM_WORKERS env variable
 function start() {
-  if (cluster.isMaster) {
+  var starter = process.argv[1].split(path.sep).pop(); //check if app is being started by a test
+  if (cluster.isMaster && starter !== '_mocha') {
```
Comment (jcesarmobile) [inline application.js]: This looks good to me, but I want more opinions about it.

File: Gruntfile.js (hunk 1 above)
Comment (rachael-oregan) [inline Gruntfile.js]: Ya ok, I left Corinne's changes untouched but I'll add this back in. Thanks

File: lib/hello.js (hunk 2 above)
Comment (rachael-oregan) [inline lib/hello.js]: Yes this gets rid of the deprecate warnings, this was a part of Corinne's changes. I can remove this and not update body-parser version?

Comment (jcesarmobile) [inline lib/hello.js]: nah, it it removes the deprecate warnings and keeps working fine it's ok to me. This template was already using a most up to date version than other templates (1.15.x vs 1.0.x) that might have the deprecate warnings already, so the other option is to downgrade it to 1.0.2 as the others. But I think it's not needed if the deprecate warnings are gone.

Comment (rachael-oregan) [inline lib/hello.js]: Cool, I'll leave as it is and just remove the commented out code as the deprecated warnings are gone

File: application.js (hunk 3 above)
Comment (sedroche) [inline application.js]: Other services have a `master-only` flag
https://github.com/feedhenry/fh-dataman/blob/master/src/app.js#L152

Can we add that flag and run the tests with it?

Comment (jcesarmobile) [inline application.js]: I just noticed that dataman is using `NODE_ENV` environment variable for running tests and check it like this `process.env.NODE_ENV === 'test'` (same line you linked)

Comment (rachael-oregan) [inline application.js]: `process.env.NODE_ENV === 'test'` works and is more readable. Adding this in. Thanks guys

Comment (jcesarmobile) [inline application.js]: You also need it on coverage_accept, and you can delete `var path = require('path');`

File: README.md [hunk 4]
```diff
@@ -24,3 +24,106 @@ This is a blank 'hello world' FeedHenry MBaaS. Use it as a starting point for bu
...
+All the tests are in the "test/" directory. The cloud app is using mocha as the test runner.
+
+### Unit and acceptance tests
+
+* all the tests:
+
+With [MongoDB](#setup-mongodb) and [Redis](#setup-redis) running
+
+```shell
+npm run test
```
Comment (jcesarmobile) [inline README.md]: To run tests you don't need run, `npm test` is enough.

**Pull request merged**